src/interface/state_descriptor.cpp

```diff
@@ -566,4 +566,37 @@ StateDescriptor::GetVariableNames(const Metadata::FlagCollection &flags) {
   return GetVariableNames({}, flags, {});
 }
 
+// Get the total length of this StateDescriptor's variables when packed
+int StateDescriptor::GetPackDimension(const std::vector<std::string> &req_names,
+                                      const Metadata::FlagCollection &flags,
+                                      const std::vector<int> &sparse_ids) {
+  std::vector<std::string> names = GetVariableNames(req_names, flags, sparse_ids);
+  int nvar = 0;
+  for (auto name : names) {
+    const auto &meta = metadataMap_[VarID(name)];
+    int var_len = 0;
+    if (meta.Shape().size() < 1) {
+      var_len = 1;
+    } else {
+      var_len = meta.Shape()[0];
+    }
+    nvar += var_len;
```

Comment:
Could this just be
`nvar += meta.Shape().size() < 1 ? 1 : meta.Shape()[0];` ?

Reply:
This was a product of special-casing different `TopologicalElements` originally.  Since now we have to special-case tensors due to below, I'm inclined to keep the extra verbosity to keep it instantly readable.

Reply:
I am going to push back against the revised branchy code. See my review.